File: problem_set/check-permutation-lcci/check-permutation-lcci.cpp

```
// @Title: 判定是否互为字符重排 (Check Permutation LCCI)
// @Author: robert.sunq
// @Date: 2020-03-22 00:25:13
// @Runtime: 4 ms
// @Memory: 7.5 MB

class Solution {
public:
    bool CheckPermutation(string s1, string s2) {
            int len_s1 =  s1.size();
            int len_s2 = s2.size();
            if(len_s1 != len_s2 ) return false;
            int mp1[260]={0};
            for(int i=0;i<len_s2;i++){
                    mp1[s1[i]]++;
                    mp1[s2[i]]--;
            }
            for(int i=0;i<256;i++){
                if(mp1[i] != 0)
                   return false;

            }
            return true;

    }
};

File: problem_set/check-permutation-lcci/check-permutation-lcci.java

```
// @Title: 判定是否互为字符重排 (Check Permutation LCCI)
// @Author: robert.sunq
// @Date: 2021-08-02 15:50:35
// @Runtime: 0 ms
// @Memory: 36.2 MB

class Solution {
    public boolean CheckPermutation(String s1, String s2) {
        int len1 = s1.length();
        int len2 = s2.length();
        // 长度不相等直接退回
        if(len1 != len2 ) return false;
        // 保存字符出现次数
        int[] dict = new int[256];
        for(int i=0;i<len1;i++){
            dict[s1.charAt(i)]++;
            dict[s2.charAt(i)]--;
        }
        for(int i=0;i<256;i++){
            if(dict[i] != 0) {
                return false;
            }
        }
        return true;

    }
}
